var/spack/repos/builtin/packages/flux-sched/package.py

```diff
@@ -64,6 +64,7 @@ class FluxSched(CMakePackage, AutotoolsPackage):
     depends_on(
         "boost+exception+filesystem+system+serialization+graph+container+regex@1.53.0,1.59.0: "
     )
+    depends_on("python@3.6:", type=("build", "run"))
```

Comment:
```suggestion
    extends("python")
```
I'm not 100% certain, but `extends("python")` looks better to me, and it should also take care of setting the path to make the build find the python interpreter.

This should use the existing support for packages that also build Python code.
The version `@3.6:` is not needed, as 3.7.0 is the oldest version that spack's python recipe can offer.

Reply:
As python can come from places other than Spack (externals), the version requirement is still useful.  In the end, I'm indifferent to `extends()` vs. `depends_on()`, but the version requirement still has a purpose.

Reply:
I tried to use `extends()`, it seems to provide `python.path` that other specs use.
And it can be used in the builder classes like so:
```py
 class AutotoolsBuilder(AutotoolsBuilder):
+    def setup_build_environment(self, env):
+        env.set("PYTHON", python.path)
```
This is the part you've to fix to pass `spack audit` in CI. 

Also, I want to hint that `flux-sched@master` fails to build for me, not finding the python setuptools (even when added as a dependency).

Anyway, I retracted my changes requested flag and leave it to other reviewers, but it can only be merged if you fix the audit error in CI.